.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,67 @@
+name: 'Unit Tests'
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  unit-tests:
+    name: 'Unit Tests'
+    runs-on: ubuntu-latest
+    if: startswith(github.head_ref, 'feat/') || startswith(github.head_ref, 'bug/') || startswith(github.head_ref, 'patch/')
+    env:
+      working-directory: ./tests/mcp
```

Comment:
I think this needs to change. Probably make `./tests` and make later overrides `${{env.working-directory}}/mcp` (or whatever)